AmmadAmjad-CodeTest/Server/endpoints/comments/routes/index.js

```diff
@@ -0,0 +1,207 @@
+    .then(function(comment){
+      if (!comment) {
+        return res.status(404).json({
+          message : "Comment not found. Please make sure the commentId is valid and comment exists"
+        });
+      }else if (comment && comment.authorId !== req.user.id) {
+        return res.status(401).json({
+          message : "Only comment author can delete the comment"
+        });
+      }else{
+        db.comment.destroy({
+          where : {
+            id: id,
+          }
+        })
+        .then(function(status){
+          res.json({
+            message : "Comment deleted successfully"
+          });
+        });
+      }
+    })
+    .catch(function(e){
+      res.status(403).json({
+        message : "Failed to delete the comment",
+        error : String(e)
+      });
+    });
+  });
+
+  // VOTE FOR THE COMMENT
+  app.post('/comment/:id/vote' , middleware.requireAuthentication , function(req , res){
+    const id = parseInt(req.params.id);
+    if (id === undefined || id === null || id <= 0) {
+      res.status(403).json({
+        message : "Please provide a valid comment iD"
+      });
+      return;
+    }
+
+    db.commentVoters.findOne({
+      where : {
+        commentId : id,
+        userId : req.user.id
+      }
+    })
+    .then(function(existing){
+      if (existing) {
+        res.status(401).json({
+          message : "You have already voted for this comment"
+        });
+      }else{
+        db.commentVoters.create({
+          commentId : id,
+          userId : req.user.id
+        })
+        .then(function(status){
+          res.json({
+            message : "Vote added to the comment successfully"
+          });
+        })
+      }
+    })
+    .catch(function(e){
+      res.status(403).json({
+        message : "Failed to add vote to the comment",
+        error : String(e)
+      });
+    });
+
+  });
+
+
+  // RECIND VOTE
+  app.delete('/comment/:id/vote' , middleware.requireAuthentication , function(req , res){
+    const id = parseInt(req.params.id);
+    if (id === undefined || id === null || id <= 0) {
+      res.status(403).json({
+        message : "Please provide a valid comment iD"
+      });
+      return;
+    }
+
+    db.commentVoters.findOne({
+      where : {
+        commentId : id,
+        userId : req.user.id
+      }
+    })
+    .then(function(existing){
+      if (!existing) {
+        res.status(401).json({
+          message : "You have never voted for this comment before"
+        });
+      }else{
+        db.commentVoters.destroy({
+          where : {
+            commentId : id,
+            userId : req.user.id
+          }
+        })
+        .then(function(status){
+          res.json({
+            message : "Vote rescinded successfully"
+          });
+        })
+      }
+    })
+    .catch(function(e){
+      res.status(403).json({
+        message : "Failed to rescind the vote",
+        error : String(e)
+      });
+    });
+
+  });
+module.exports = app;
```

Comment:
Sorry to be nitpick, seeing some files having a line break on last line some files don't, better keep them consistent. Suggesting all files have a line break for their last line just like any other lines.